### PR TITLE
Issue/2652 unobscure password post options

### LINKED ISSUE #2652: Add unobscure button post passwords.
Posts edited in the app and password protected, have their passwords obscured now.  It would be nice to have an unobscure button to be able to see the password.

[Here's a short recording](https://cloudup.com/cCxCu6hpdPL) explaining the idea.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Fixes #2652.

Also fixes an issue that was causing the password to be reset when the device was rotated.  This can be reproduce in develop adding a password to a post (as mentioned in the steps below) and then rotating the device.

**To test:**

- Create a new post in the app.
- Go to the post options (top-right corner "..." then select "Options").
- Under the visibility option pick: "Password Protected".
- Enter a password.
- Show and hide it using the added button.

### CODE REVIEW

File: WordPress/Classes/System/Constants.m [hunk 1]
```diff
@@ -24,6 +24,11 @@
 NSString *const WPBlavatarBaseURL                                   = @"http://gravatar.com/blavatar";
 NSString *const WPGravatarBaseURL                                   = @"http://gravatar.com/avatar";
 
+/// Image file names
+///
+NSString *const WPIconShowPassword                                  = @"icon-secure-text-visible";
```
Comment (frosty) [inline WordPress/Classes/System/Constants.m]: Can we use the `visible` and `not-visible` Gridicons here instead of images? Or are they not the same as the ones we're using?

File: WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m [hunk 2]
```diff
@@ -584,25 +613,44 @@ - (UITableViewCell *)configureMetaPostMetaCellForIndexPath:(NSIndexPath *)indexP
...
+    
+    [self configureVisibilityButtonForPasswordCell:textCell];
+    
+    return textCell;
+}
+
+- (void)configureVisibilityButtonForPasswordCell:(WPTextFieldTableViewCell *)textCell
+{
+    self.passwordVisibilityButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    self.passwordVisibilityButton.frame = CGRectMake(0, 0, 24, 20);
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m]: Can we pass the image size here instead of using magic numbers? Or will the button size to fit its image anyway?

**Comment by frosty** [comment]
Hey @diegoreymendez! Changes look good, except two small comments and: the button state is the opposite of what we use on the login screen. So on login, if the button shows the open eye then the password is visible. If it shows the closed eye, the password is hidden. I think that way of doing it makes the most sense.

File: WordPress/Classes/System/Constants.m (hunk 1 above)
Comment (diegoreymendez) [inline WordPress/Classes/System/Constants.m]: I can check them out and remove the pre-existing local image in the process.

**Comment by diegoreymendez** [comment]
@frosty - Good feedback.

These are my changes to address your comments:

- Changed the icons to load from Gridicons.
- Removed the constants I had added.
- Changed `WPWalkthroughTextField.m` to also use the Gridicons icons.  You can test this by trying to add a self-hosted site to the app.
- Deleted the local images we had for those two icons.

So the PR is ready for round 2.  Let me know.

File: WordPress/Classes/ViewRelated/NUX/WPWalkthroughTextField.m [hunk 3]
```diff
@@ -57,8 +59,8 @@ - (void)commonInit
     self.showTopLineSeparator = NO;
     self.showSecureTextEntryToggle = NO;
 
-    self.secureTextEntryImageVisible = [UIImage imageNamed:WPIconShowPassword];
-    self.secureTextEntryImageHidden = [UIImage imageNamed:WPIconHidePassword];
+    self.secureTextEntryImageVisible = [Gridicon iconOfType:GridiconTypeVisible];
+    self.secureTextEntryImageHidden = [Gridicon iconOfType:GridiconTypeNotVisible];
 
     self.secureTextEntryToggle = [UIButton buttonWithType:UIButtonTypeCustom];
     self.secureTextEntryToggle.frame = CGRectMake(0, 0, 40, 30);
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/NUX/WPWalkthroughTextField.m]: Can you also remove the manual frame setting here?

**Comment by frosty** [comment]
Changes look good, @diegoreymendez! One final tweak (in addition to the code comment I just added): 

t looks like the button needs to have a different `tintColor` set – the original image must've been colored instead of tinted. I think `greyLighten10` should do it (needs to be set for both sign in and the post settings).

**Comment by diegoreymendez** [comment]
Adjusted the tint color.  Ready for another look.

File: WordPress/Classes/ViewRelated/NUX/WPWalkthroughTextField.m (hunk 3 above)
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/NUX/WPWalkthroughTextField.m]: Considering this is outside the scope of the PR, can you specify the change you'd like here?  I'd like to avoid a trial and error.

If I adjust the button to the default Gridicon size, the new code is going to have a different size.

PS: I don't think it'd be wise to continue extending this PR, but I don't mind changing it if the request is specific.

**Comment by frosty** [comment]
:shipit:!

**Comment by diegoreymendez** [comment]
Thanks James!
